### PR TITLE
[Merged by Bors] - feat: any function is integrable on a finset wrt to a finite measure

### PR DESCRIPTION
**Pull request by YaelDillies**
Also restate `integral_finset` in terms of `IntegrableOn`
From PFR

---

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

File: Mathlib/MeasureTheory/Integral/Bochner.lean [hunk 1]
```diff
@@ -1761,7 +1762,7 @@ theorem integral_countable [MeasurableSingletonClass α] (f : α → E) {s : Set
   simp
 
 theorem integral_finset [MeasurableSingletonClass α] (s : Finset α) (f : α → E)
-    (hf : Integrable f (μ.restrict s)) :
+    (hf : IntegrableOn f s μ) :
```
Comment (kex-y) [inline Mathlib/MeasureTheory/Integral/Bochner.lean:1765]: Can you make this change for the lemma above as well

File: Mathlib/MeasureTheory/Integral/IntegrableOn.lean [hunk 2]
```diff
@@ -188,6 +188,11 @@ theorem integrableOn_finite_iUnion [Finite β] {t : β → Set α} :
   cases nonempty_fintype β
   simpa using @integrableOn_finset_iUnion _ _ _ _ _ f μ Finset.univ t
 
+lemma IntegrableOn.finset [MeasurableSingletonClass α] {μ : Measure α} [IsFiniteMeasure μ]
+    {s : Finset α} {f : α → E} : IntegrableOn f s μ := by
```
Comment (b-mehta) [inline Mathlib/MeasureTheory/Integral/IntegrableOn.lean:192]: Similar to the earlier ones, let's add something like this:
```
lemma _root_.Set.Finite.integrableOn [MeasurableSingletonClass α] {μ : Measure α}
    [IsFiniteMeasure μ] {s : Set α} (hs : s.Finite) {f : α → E} : IntegrableOn f s μ := by
  simpa using IntegrableOn.finset (s := hs.toFinset)
```

**Comment by b-mehta** [comment]
Thanks!

bors merge

**Pull request closed without merging**